src/layers/XRLayer/XRLayer.js

```diff
@@ -169,12 +171,30 @@ export default class XRLayer extends Layer {
   draw({ uniforms }) {
     const { textures, model } = this.state;
     if (textures && model) {
-      const { sliderValues, colorValues, opacity } = this.props;
+      const {
+        sliderValues,
+        colorValues,
+        opacity,
+        domain,
+        dtype,
+        channelIsOn
+      } = this.props;
+      // Check number of textures not null.
+      const numTextures = Object.values(textures).filter(t => t).length;
+      // Slider values and color values can come in before textures since their data is async.
+      // Thus we pad based on the number of textures bound.
+      const { paddedSliderValues, paddedColorValues } = padColorsAndSliders({
+        sliderValues: sliderValues.slice(0, numTextures),
+        colorValues: colorValues.slice(0, numTextures),
+        channelIsOn: channelIsOn.slice(0, numTextures),
```

Comment:
I think `padColorAndSliders` is pretty lightweight and this makes sense in `draw` to avoid the kinds of race conditions.

Reply:
Just thinking about this a little more, and I don't want to waste your time with this but padding every draw call feels unecessary. Do you think there is a way to check if `slidersValues` !== `numTextures` and then not set setUniforms if that's the case (or pad?).

Reply:
I'm not sure I understand how a race condition could occur - could you elaborate a bit?  If I understand the second half of what you're saying, we should call `draw` but not `setUniforms`?  Then there would still be a rendering but we would have no information so WebGL would draw whatever its various defaults are.  My guess is that this would cause some flickering, but I'll try it.

Reply:
Yes, this causes flickering when you add a channel.

Reply:
Ah no, I tried something different, where there is no draw call.  I'll test what you said a bit - it seems to work.

Reply:
Unfortunately, it does not work.  The `StaticImageLayer` flickers on channel addition and removal.  This effect could be obscured on the TileLayer by its caching but it's hard to tell.